crates/net/eth-wire/src/types/transactions.rs

```diff
@@ -59,9 +61,20 @@ impl PooledTransactions {
     }
 }
 
-impl From<Vec<TransactionSigned>> for PooledTransactions {
-    fn from(txs: Vec<TransactionSigned>) -> Self {
-        PooledTransactions(txs.into_iter().map(Into::into).collect())
+impl TryFrom<Vec<TransactionSigned>> for PooledTransactions {
+    type Error = TransactionConversionError; // Assume this is defined elsewhere
+
+    fn try_from(txs: Vec<TransactionSigned>) -> Result<Self, Self::Error> {
+        let mut pooled_txs = Vec::new();
+
+        for tx in txs {
+            match PooledTransactionsElement::try_from(tx) {
+                Ok(pooled_tx) => pooled_txs.push(pooled_tx),
+                Err(e) => return Err(e),
+            }
+        }
+
+        Ok(PooledTransactions(pooled_txs))
```

Comment:
you can keep the previous impl and replace the `Into::into` because collect also supports collecting into Result vec